File: web/src/main/resources/components/src/app/util/Player.ts
/**
 * Created by Jakub on 21.05.2017.
 */
export class Player {
    name: string;
    color: boolean;
    type: string;


    constructor(name: string, color: boolean, type: string) {
        this.name = name;
        this.color = color;
        this.type = type;
    }


    toJson(): string {
        return JSON.stringify(this);
    }
}

File: web/src/main/java/si/ai/othello/web/game/player/Player.java
package si.ai.othello.web.game.player;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NonNull;
import si.ai.othello.game.player.IPlayer;
import si.ai.othello.game.player.PlayerType;

/**
 * @author Jakub Licznerski
 *         Created on 21.05.2017.
 */
@Data
@AllArgsConstructor
public class Player {
    @NonNull private String name;
    private boolean color;
    private PlayerType type;

    public Player(IPlayer player) {
        this.name = player.getName();
        this.color = player.getColor();
        this.type = player.getPlayerType();
    }
}
